Submitted By:            Igor Živković <contact@igor-zivkovic.from.hr>
Date:                    2014-12-22
Initial Package Version: 3.18.1
Upstream Status:         Not submitted
Origin:                  Self
Description:             Fixes build with BusyBox ash shell.

diff -Naur linux-3.18.1.orig/arch/x86/kernel/cpu/mkcapflags.sh linux-3.18.1/arch/x86/kernel/cpu/mkcapflags.sh
--- linux-3.18.1.orig/arch/x86/kernel/cpu/mkcapflags.sh	2014-12-16 18:39:45.000000000 +0100
+++ linux-3.18.1/arch/x86/kernel/cpu/mkcapflags.sh	2014-12-22 13:47:00.238970516 +0100
@@ -6,7 +6,7 @@
 IN=$1
 OUT=$2
 
-function dump_array()
+dump_array()
 {
 	ARRAY=$1
 	SIZE=$2
@@ -28,7 +28,7 @@
 		# If the /* comment */ starts with a quote string, grab that.
 		VALUE="$(echo "$i" | sed -n 's@.*/\* *\("[^"]*"\).*\*/@\1@p')"
 		[ -z "$VALUE" ] && VALUE="\"$NAME\""
-		[ "$VALUE" == '""' ] && continue
+		[ "$VALUE" = '""' ] && continue
 
 		# Name is uppercase, VALUE is all lowercase
 		VALUE="$(echo "$VALUE" | tr A-Z a-z)"
diff -Naur linux-3.18.1.orig/scripts/link-vmlinux.sh linux-3.18.1/scripts/link-vmlinux.sh
--- linux-3.18.1.orig/scripts/link-vmlinux.sh	2014-12-16 18:39:45.000000000 +0100
+++ linux-3.18.1/scripts/link-vmlinux.sh	2014-12-22 13:44:05.500155051 +0100
@@ -111,7 +111,7 @@
 }
 
 # Delete output files in case of error
-trap cleanup SIGHUP SIGINT SIGQUIT SIGTERM ERR
+trap cleanup SIGHUP SIGINT SIGQUIT SIGTERM
 cleanup()
 {
 	rm -f .old_version
